src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,194 @@
+                    return;
+                default:
+                    if (input.substring(0, 4).equals("todo")) {
+                        try {
+                            String description = input.substring(5);
+                            todo.add(new Task(description));
+                            System.out.println("____________________________________________________________");
+                            System.out.println("Got it. I've added this task: ");
+                            System.out.println(todo.get(todo_index).toString());
+                            save += todo.get(todo_index).toString()+"\n";
+                            writeToFile("data/duke.txt",save);
+                            todo_index += 1;
+                        } catch (StringIndexOutOfBoundsException e) {
+                            System.out.println("____________________________________________________________");
+                            System.out.println("OOPS!!! The description of a todo cannot be empty.");
+                            System.out.println("____________________________________________________________");
+                        }
+                    } else if (input.startsWith("deadline")) {
+                        int descriptionIndex = 9;
+                        String task = "deadline";
+                        try {
+                            ExceptionTry(task, input);
+                            String description = input.substring(input.indexOf("deadline") + descriptionIndex, input.indexOf("/by") - 1);
+                            String by = input.substring(input.indexOf("/by") + 4);
+                            todo.add(new Deadline(description, by));
+                            System.out.println("____________________________________________________________");
+                            System.out.println("Got it. I've added this task: ");
+                            System.out.println(todo.get(todo_index).toString());
+                            save += todo.get(todo_index).toString()+"\n";
+                            writeToFile("data/duke.txt",save);
+                            todo_index += 1;
+                        } catch (DukeException e) {
+                            System.out.println("____________________________________________________________");
+                            System.out.println("OOPS!!! The description of a deadline must have /by.");
+                            System.out.println("____________________________________________________________");
+                        } catch (StringIndexOutOfBoundsException e) {
+                            System.out.println("____________________________________________________________");
+                            System.out.println("OOPS!!! The description of a deadline cannot be empty.");
+                            System.out.println("____________________________________________________________");
+                        }
```

Comment:
Making switch case statements this long makes the code extremely long and hard to read. My suggestion would be to abstract this logic into a separate function. This way you wouldn't have to repeat the same code multiple times in line with the DRY (Don't Repeat Yourself) principle